### ISSUE TITLE
Google Chrome warning: [Deprecation] Synchronous XMLHttpRequest

### ISSUE THREAD

**Issue by feuzeu** [jaxon-php/jaxon-js#4]
_From @jatubio on May 24, 2017 13:4_

I'm getting this message in Google Chrome, on Developers Tools Log:

[Deprecation] Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org/.

On: jaxon.core.min.js:1

_Copied from original issue: jaxon-php/jaxon-core#20_

**Comment by jatubio** [comment]
With 2.0 version I'm not getting this warning any more.

**Comment by jatubio** [comment]
> With 2.0 version I'm not getting this warning any more.

It was because all my requests were asynchronous. On synchronous request the message continues to appear.

**Comment by feuzeu** [comment]
Hi,
I would like to share some ideas about a possible solution to this issue.
What do you think about re-implementing synchronous requests with a queueing system? That is, the requests that are supposed to be synchronous are deferred, queued, and executed one after another with no concurrency.
The js library usage will need some changes, since the call to a Jaxon js function will not be able to return the value returned by the PHP function anymore. The developer will need to define a callback to get it.

Other questions will arise on how to handle the case where synchronous and asynchronous requests are made simultaneously.
Shall the asynchronous requests be deferred and ran after the synchronous ones?

**Comment by feuzeu** [comment]
The following library can be used to implement request queueing: https://github.com/d3/d3-queue.

**Comment by feuzeu** [comment]
Hi all,

I've reimplemented the synchronous request feature using a queue.
The feature is available on the develop branch, and still in beta stage.
https://github.com/jaxon-php/jaxon-js/tree/develop
I hope you will be able to test it against various scenarii and provide feedbacks.

That's how it works.
When the user sends a synchronous request, Jaxon will send it asynchronously, but all the requests that will be sent and all the responses that will be receivend afterwards will be queued, and processed only after the response to the synchronous request is received and processed. Nothing happens in the interval.

So the library will still behave like before, excepted that the synchronous javascript function will not return the same value as the PHP function anymore.

**Comment by feuzeu** [comment]
Fixed release v3.2.0.
